### PR TITLE
Clarify that RP2350 gets its unique id from OTP

### LINKED ISSUE #1915: pico_unique_id header needs some updates
From reading the comments in [unique_id.h](https://github.com/raspberrypi/pico-sdk/blob/develop/src/rp2_common/pico_unique_id/include/pico/unique_id.h) (which also appear [here](https://www.raspberrypi.com/documentation/pico-sdk/high_level.html#group_pico_unique_id) ) you might think that `pico_get_unique_board_id` **always** returns the ID of the flash chip. However looking at the [implementation](https://github.com/raspberrypi/pico-sdk/blob/develop/src/rp2_common/pico_unique_id/unique_id.c#L15) it looks like that on RP2350 it actually gives you the `SYS_INFO_CHIP_INFO` (which I think is described in section 5.4.8.17. of https://datasheets.raspberrypi.com/rp2350/rp2350-datasheet.pdf ?).
And I guess this also means that you _won't_ get a unique_id of all-0xEE on PICO_NO_FLASH builds for RP2350? :wink:

### PR DESCRIPTION
**Pull request by peterharperuk**
Fixes #1915

### CODE REVIEW

**Comment by lurch** [comment]
Looks good, but I wonder if this ought to use the `\if` mechanism that can be seen in e.g. https://github.com/raspberrypi/pico-sdk/pull/1945/files ?

**Comment by peterharperuk** [comment]
yes, looks like that might be a good idea.

File: src/rp2_common/pico_unique_id/include/pico/unique_id.h [hunk 1]
```diff
@@ -64,11 +83,16 @@ void pico_get_unique_board_id(pico_unique_board_id_t *id_out);
 /*! \brief Get unique ID in string format
  *  \ingroup pico_unique_id
  *
- * Get the unique 64-bit device identifier which was retrieved from the
- * external NOR flash device or OTP memory at boot, formatted as an ASCII hex string.
- * Will always 0-terminate.
+ * Get the unique 64-bit device identifier formatted as a 0-terminated ASCII hex string.
```
Comment (lurch) [inline src/rp2_common/pico_unique_id/include/pico/unique_id.h:86]: Would "null-terminated" be better than "0-terminated" ?

**Comment by lurch** [comment]
One small quibble but otherwise looks great! :+1:

**Review by kilograham — APPROVED** [review]

**Pull request merged**